R/create.R

```diff
@@ -182,4 +183,91 @@ create_initial_conditions <- function(data, delays, rt_prior, generation_time, m
   }
   
   return(init_fun)
+}
+
+
+
+#' Create a List of Stan Arguments
+#'
+#'
+#' @description Generates a list of arguments as required by `rstan::sampling` (when `method = "exact`) or 
+#' `rstan::vb` (when `method = "approximate`). See `create_stan_args()` for the defaults and the relevant `rstan`
+#' functions for additonal options.
+#' @param model A stan model object, defaults to packaged model if not supplied.
+#' @param data A list of stan data as created by `create_stan_data`
+#' @param init Initial conditions passed to `rstan`. Defaults to "random" but can also be a function (
+#' as supplied by `create_intitial_conditions`).
+#' @param samples Numeric, defaults to 1000. The overall number of posterior samples to return (Note: not the 
+#' number of samples per chain as is the default in stan).
+#' @param stan_args A list of stan arguments to be passed to `rstan::sampling` or `rstan::vb` (when using the "exact"
+#' or "approximate" method). For `method = approximate` an additional argument `trials` indicates the number of attempts to make 
+#' using variational inference before returning an error (as stochastic failure is possible). The default for this is 5.
+#' @param method A character string defaults to "exact". Also accepts "approximate". Indicates the fitting method to be used
+#' this can either be "exact" (NUTs sampling) or "approximate" (variational inference). The exact approach returns samples
+#' from the posterior whilst the approximate method returns approximate samples. The approximate method is likely to return results 
+#' several order of magnitudes faster than the exact method.
+#' @param verbose Logical, defaults to `FALSE`. Should verbose progress messages be returned.
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' 
+#' # default settings
+#' create_stan_args()
+#' 
+#' # approximate settings
+#' create_stan_args(method = "approximate") 
+#' # increasing warmup
+#' create_stan_args(stan_args = list(warmup = 1000))
+create_stan_args <- function(model, data = NULL, init = "random", 
```

Comment:
I am not sure this is the way to go for a few reasons. Firstly this would introduce a lot of structure and I am not sure the typical user (a public health professional) is going to know what to do with an R6 object. Secondly, does this approach mean that we would have to provide defaults for all of the `stan` arguments which I think this would require? 

I am happy to see an alternative implementation here but I think as another PR.